platform/security/encryption/platform-security-encryption-crypter/src/main/java/ddf/security/encryption/crypter/Crypter.java

```diff
@@ -0,0 +1,277 @@
+ * underlying cryptography implementations for use throughout the system.
+ */
+public class Crypter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(Crypter.class);
+  private static final String DEFAULT_KEYSET_FILE_NAME = "default";
+  private static final String KEYSET_FILE_EXTENSION = ".json";
+  private static final int ASSOCIATED_DATA_BYTE_SIZE = 10;
+
+  private final String keysetDir =
+      AccessController.doPrivileged(
+          (PrivilegedAction<String>) () -> System.getProperty(SecurityConstants.KEYSET_DIR));
+  private final String associatedDataPath =
+      AccessController.doPrivileged(
+          (PrivilegedAction<String>)
+              () -> System.getProperty(SecurityConstants.ASSOCIATED_DATA_PATH));
+
+  @VisibleForTesting KeysetHandle keysetHandle;
+  private byte[] associatedData;
+  private Aead aead;
+
+  /**
+   * Read/create a specified keyset under the keyset directory.
+   *
+   * @param keysetFileName name of the keyset to read/create.
+   * @throws CrypterException
+   */
+  public Crypter(String keysetFileName) throws CrypterException {
+    String keysetLocation = Paths.get(keysetDir, keysetFileName + KEYSET_FILE_EXTENSION).toString();
+
+    synchronized (Crypter.class) {
+      File keysetFile = new File(keysetLocation);
+
+      try {
+        AeadConfig.register();
+        if (!keysetFile.exists()) {
+          keysetHandle = initKeysetHandle(keysetFile);
+        } else {
+          keysetHandle = readKeysetHandle(keysetFile);
+        }
+        aead = AeadFactory.getPrimitive(keysetHandle);
+        associatedData = getAssociatedData(keysetHandle);
+      } catch (GeneralSecurityException | IOException e) {
+        LOGGER.warn("Problem initializing keyset.");
+        throw new CrypterException("Problem initializing keyset", e);
+      }
+    }
+  }
+
+  public Crypter() throws CrypterException {
+    this(DEFAULT_KEYSET_FILE_NAME);
+  }
+
+  /**
+   * Encrypts a plain text value using Tink.
+   *
+   * @param plainTextValue The value to encrypt.
+   */
+  public String encrypt(String plainTextValue) throws CrypterException {
+    if (plainTextValue == null || plainTextValue.isEmpty()) {
+      throw new CrypterException("Value to encrypt cannot be null or empty.");
+    }
+    if (associatedData == null) {
+      throw new CrypterException("Associated data cannot be null.");
+    }
+    try {
+      byte[] encryptedBytes = aead.encrypt(plainTextValue.getBytes(), associatedData);
+      return Base64.getEncoder().encodeToString(encryptedBytes);
+    } catch (GeneralSecurityException e) {
+      LOGGER.debug("Problem encrypting.", e);
+      throw new CrypterException("Problem encrypting.", e);
+    }
+  }
+
+  /**
+   * Decrypts a plain text value using Tink
+   *
+   * @param encryptedValue The value to decrypt.
+   */
+  public String decrypt(String encryptedValue) throws CrypterException {
+    if (encryptedValue == null || encryptedValue.isEmpty()) {
```

Comment:
❓ Is there a reason that Apache commons can't be included in this module (https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isBlank-java.lang.CharSequence-)

Reply:
After adding the apache commons dependency to the crypter class, `platform-osgi-configadmin` failed to resolve due to not being able to find the apache commons packages.

Since the `platform-osgi-configadmin` bundle is so finicky when it comes to imports, I opted to use my own version of `isBlank()`:
```
string == null || string.trim().isEmpty()
```

Reply:
Gotcha. 👍